### ISSUE TITLE
Request a resource

### ISSUE THREAD

**Issue by fosterlynn** [gopacifia/DEEP#8]
Request a resource (create a customer order or an exchange from a resource page).

This could be something like adding a button to the resource page that starts a customer order, prefilling it.

Or it could be cleaner than that, a new feature.

The current customer order function needs work anyhow, so we should take that into consideration in deciding if to code something new.

Note this could be connected to the search feature #7

**Comment by bhaugen** [comment]
Skill level: difficult. Requires understanding of Django and the existing DEEP model and code. Also requires some architectural, server-side, user experience, and user interface design skills. Probably will want some javascript goodness on the UI.

Steps:
* Decide whether to modify [the existing create_order code](https://github.com/gopacifia/DEEP/blob/master/valuenetwork/valueaccounting/views.py#L3362) or not.
   * The existing code does not work very well anymore because nobody has been using it, so if somebody wants to start by modifying that code, I'll make it work again first. If somebody assigns themselves to this issue and wants to modify the existing code, I'll add a separate issue to make the existing code work again and I'll assign myself to that. Others can help if they want.)
* Design the feature. This will require a bunch of upfront design which should be a collaborative effort.
* Will suggested Test-Driven Development. Here's [the test for the current version](https://github.com/gopacifia/DEEP/blob/master/valuenetwork/valueaccounting/tests/test_orders.py), which does not pass anymore, and has been commented out in [the test list](https://github.com/gopacifia/DEEP/blob/master/valuenetwork/valueaccounting/tests/__init__.py).

**Comment by bhaugen** [comment]
I'm rethinking this, and recommend starting over - not modifying create_order. If all you want to do with this feature is order something that exists in inventory, it's a lot simpler. Create_order also handles the case where you want to order something that does not exist yet and needs to be produced. That's a lot harder.

**Comment by numapanuma** [comment]
I think the feature of ordering something that exists in inventory is enough. But there should be a way that suppliers of a type of resource get notified when a customer wanted to order and couldn't because there was no quantity available. The main difference is when an order of existing resource is made, there is a specific supplier related to the resource. Whereas when ordering a resource not available should send a quote request to all suppliers that have subscribed to the resource type or corresponding category.

**Comment by fosterlynn** [comment]
>I think the feature of ordering something that exists in inventory is enough. But there should be a way that suppliers of a type of resource get notified when a customer wanted to order and couldn't because there was no quantity available. 

Thinking about this a bit more, I think this and Search Inventory #7 need to be designed together.  And this will have a button somewhere on the new search result page.  

In terms of suppliers being notified, that means that #7 needs to support zero quantity resources that are offered but out of stock.  And for that matter, some groups may want to make to order, and not stock anything.

**Comment by numapanuma** [comment]
Or we could simply put an Order button in the resource page. So after searching inventory, you click on the resource you want to open the resource page, and from there you can click on the order button. It would make sense that you double check resource info before ordering it don't you think?

> In terms of suppliers being notified, that means that #7 needs to support zero quantity resources that are offered but out of stock. And for that matter, some groups may want to make to order, and not stock anything.

yes!

**Comment by fosterlynn** [comment]
>Or we could simply put an Order button in the resource page.

If we require there to always be a resource.  Let's see how the new search function works, and then see.  In the meantime, fine to put on the resource page, certainly won't be a waste.

**Comment by numapanuma** [comment]
> If we require there to always be a resource.
That's right, a resource type could not have a resource yet. Should customers be able to say "hey, I'd like a new resource that has never been offered here", or even a new resource type?
Ok so le'ts focus on the new search function first #7 .

**Comment by fosterlynn** [comment]
>Ok so le'ts focus on the new search function first #7 .

:+1:
